compiler/rustc_middle/src/ty/sty.rs

```diff
@@ -2801,7 +2801,13 @@ impl<'tcx> Ty<'tcx> {
     /// bound such as `[_]: Copy`. A function with such a bound obviously never
     /// can be called, but that doesn't mean it shouldn't typecheck. This is why
     /// this method doesn't return `Option<bool>`.
+    #[inline]
     pub fn is_trivially_sized(self, tcx: TyCtxt<'tcx>) -> bool {
+        #[inline(never)]
```

Comment:
We have `#[cold]` as well but I never know when to use it...

Is it that cold though? It is called for all ADTs.